### PR TITLE
Observe HTTP headers for all requests

### LINKED ISSUE #1230: Observe HTTP header option for all requests
The existing `--http-header` option is currently restricted to GraphQL introspection, but it would be useful for any HTTP request quicktype makes. For example, requesting raw JSON schema from a private GitHub repository (requiring an `Authorization: token X` header).

While it might be nice to specify exactly which requests get headers and which do not, the configuration overhead would be significant. I recommend attaching specified HTTP headers to all requests.

I will have a PR up for this shortly.

### PR DESCRIPTION
**Pull request by benquarmby**
Resolves #1230

### CODE REVIEW

File: src/quicktype-core/input/JSONSchemaStore.ts [hunk 1]
```diff
@@ -3,6 +3,8 @@ import { StringMap, assert } from "../support/Support";
 export type JSONSchema = StringMap | boolean;
 
 export abstract class JSONSchemaStore {
+    constructor(protected headers?: string[]) {}
```
Comment (benquarmby) [inline src/quicktype-core/input/JSONSchemaStore.ts]: I could see moving this to the `FetchingJSONSchemaStore` constructor. Might be better there.

Comment (benquarmby) [reply src/quicktype-core/input/JSONSchemaStore.ts]: Done.

**Comment by schani** [comment]
This is failing TSLint.  The buildkite pipeline is now public - you should be able to see the errors.

**Comment by benquarmby** [comment]
Thanks, will take a look.

File: src/cli/NodeIO.ts [hunk 2]
```diff
@@ -9,12 +9,38 @@ import { messageError, JSONSchemaStore, JSONSchema, parseJSON } from "../quickty
...
+
+function parseHeaders(httpHeaders?: string[]): HttpHeaders {
+    if (!Array.isArray(httpHeaders)) {
+        return {};
+    }
+
+    return httpHeaders.reduce(
+        function (result: HttpHeaders, httpHeader: string) {
+            if (httpHeader !== undefined && httpHeader.length > 0) {
+                const [key, value] = httpHeader.split(":");
```
Comment (schani) [inline src/cli/NodeIO.ts]: This will not work if the input string is, say `Header-With-Colons: Foo:Bar:Baz`.  Maybe that's not allowed?

Comment (benquarmby) [reply src/cli/NodeIO.ts]: Yes you're right, this is a bit naive. I'll take another crack at making it more robust.

Comment (benquarmby) [reply src/cli/NodeIO.ts]: Updated, let me know what you think @schani.

Comment (schani) [reply src/cli/NodeIO.ts]: Looks good!

**Comment by schani** [comment]
Sorry, we're having some build server issues at the moment, which might or might not be our fault.

**Comment by schani** [comment]
Thank you!

**Comment by benquarmby** [comment]
🎉 

I'm using this feature from my local build right now. What is the cadence for rolling out updates to NPM?

**Comment by schani** [comment]
The cadence is it was published to NPM 50 minutes ago ;-)

https://www.npmjs.com/package/quicktype

**Comment by benquarmby** [comment]
Booyah! Thank you very much.

**Pull request merged**